gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/PropertyList.java

```diff
@@ -202,6 +202,38 @@ public void set(String key, Object value) {
   }
 
   /**
+   * Removes the property of the given key from the list.
+   *
+   * @param key property key
+   * @return true if the list contained a property with the given key
+   */
+  public boolean remove(String key) {
+    for (Property property : properties) {
+      if (property.getKey().equals(key)) {
+        return properties.remove(property);
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Removes the given property from the list.
+   *
+   * @param property property
+   * @return true if the list contained the given property
+   * *
+   *
+   */
+  public boolean remove(Property property) {
```

Comment:
I am not sure about this one. The data model allows multiple properties with the same key but different values. I guess in that case, you should use Property.equals() as this compares key **and** value. This should also be tested